### ISSUE TITLE
Regression: local installs fail with conda 4.1.5 / 4.1.6

### ISSUE THREAD

**Issue by aldanor** [conda/conda#2987]
In 4.1.6, it looks like `normalize_urls` is expecting a list and receives a dict:

```console
>>> conda.misc.explicit([pkg], env)
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)

conda/misc.py in explicit(specs, prefix, verbose, force_extract, index_args, index)
    124         index_args['prepend'] = False
    125         index_args['channel_urls'] = channels
--> 126         index.update(get_index(**index_args))
    127
    128     # Finish the MD5 verification

conda/api.py in get_index(channel_urls, prepend, platform, use_local, use_cache, unknown, offline, prefix)
     20     if use_local:
     21         channel_urls = ['local'] + list(channel_urls)
---> 22     channel_urls = normalize_urls(channel_urls, platform, offline)
     23     if prepend:
     24         channel_urls.extend(get_channel_urls(platform, offline))

conda/config.py in normalize_urls(urls, platform, offline_only)
    248     newurls = []
    249     while urls:
--> 250         url = urls[0]
    251         urls = urls[1:]
    252         if url == "system" and rc_path:

KeyError: 0
> conda/config.py(250)normalize_urls()
    249     while urls:
--> 250         url = urls[0]
    251         urls = urls[1:]

ipdb> urls
{'file://<...>/linux-64/': ('file://<...>', 0)}
```

(<...> is a long path to the package location)

In 4.1.5, file URLs are broken (which should have been fixed in 4.1.6 IIRC):

```console
HTTPError: 404 Client Error: None for url: file:///<...>/linux-64/repodata.json.bz2

During handling of the above exception, another exception occurred:
...
```

**Comment by mcg1969** [comment]
I'm sorry you got burned by this: we've addressed it in https://github.com/conda/conda/pull/2976

**Comment by aldanor** [comment]
I see, thanks. Looking forward to 4.1.7 asap then, this is a showstopper for our build system :panda_face:

**Comment by kalefranz** [comment]
fixed in #2964.  4.1.7 should be out in an hour or so
